### ISSUE TITLE
Model not meant for training?

### ISSUE THREAD

**Issue by alontalmor** [allenai/allennlp-bert-qa-wrapper#9]
Hi, 

Thanks for this wrapper,
I've been trying to use it for a different AI2 effort. I've noticed that in the forward :

        if self.training:
            loss = torch.sum(start_logits) * 0.0
            output_dict["loss"] = loss

Is this model not intended for training?

thanks!
Alon

**Comment by pdasigi** [comment]
Hi @alontalmor! I made this repository because I wanted to quickly get a demo of the BERT QA model running for something, and didn't really need to train the model. But it should be easy to get it to train too. I had been meaning to get to it some time next week. But if you need it sooner, and would like to take a stab at it yourself, please feel free to do so, and a PR would be greatly appreciated.
What needs to be done is that the 
[`text_to_instance`](https://github.com/allenai/allennlp-bert-qa-wrapper/blob/master/pretrained_bert/dataset_reader.py#L89) method in the dataset reader should be modified to also return `start_positions` and `end_positions` fields. That can be done by copying the relevant bits fo code from [here](https://github.com/huggingface/pytorch-pretrained-BERT/blob/master/examples/run_squad.py#L947-L1009). I think that should most likely be sufficient to get things running. It would be nicer to also use AllenNLP's indexers instead of manually doing all of that [here](https://github.com/allenai/allennlp-bert-qa-wrapper/blob/master/pretrained_bert/dataset_reader.py#L143-L186).

**Comment by alontalmor** [comment]
I see, thanks for the elaborate reply. I'll be working on a project with different requirements, so using this project as inspiration but not directly ... but this is very helpful, thanks for providing this!
